src/main/java/redis/clients/jedis/BuilderFactory.java

```diff
@@ -555,6 +592,143 @@ public String toString() {
     }
   };
 
+  public static final Builder<StreamInfo> STREAM_INFO = new Builder<StreamInfo>() {
+    @Override
+    @SuppressWarnings("unchecked")
+    public StreamInfo build(Object data) {
+      if (null == data) {
+        return null;
+      }
+
+      Map<String,Builder> mappingFunctions = new HashMap<>();
```

Comment:
You're creating this map during every object creation. Create it once and use rest of the time.